src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1661 @@
+            fileChooseCtx->unk_2447E[fileChooseCtx->unk_244A6] =
+                (gSaveContext.save.inventory.questItems & 0xF0000000) >> 0x1C;
+        }
 
+        // clear buffer
+        bzero(*sramCtx->saveBuf, sizeof(*sramCtx->saveBuf));
+        // read to buffer
+        func_80185968(*sramCtx->saveBuf, D_801C67C8[fileChooseCtx->unk_2448E * 2],
+                      D_801C67F0[fileChooseCtx->unk_2448E * 2]);
+
+        if (1) {}
+        func_80185968(&(*sramCtx->saveBuf)[0x2000], D_801C67C8[fileChooseCtx->unk_2448E * 2 + 1],
+                      D_801C67F0[fileChooseCtx->unk_2448E * 2 + 1]);
+        if (1) {}
+
+        // copy buffer to save context
+        Lib_MemCpy(&gSaveContext.save, sramCtx->saveBuf, sizeof(Save));
+
+        fileChooseCtx->unk_2440C[fileChooseCtx->unk_244A6] = gSaveContext.save.playerData.deaths;
+
+        for (i = 0; i < ARRAY_COUNT(gSaveContext.save.playerData.playerName); i++) {
+            fileChooseCtx->unk_24414[fileChooseCtx->unk_244A6][i] = gSaveContext.save.playerData.playerName[i];
+        }
+
+        fileChooseCtx->healthCapacity[fileChooseCtx->unk_244A6] = gSaveContext.save.playerData.healthCapacity;
+        fileChooseCtx->health[fileChooseCtx->unk_244A6] = gSaveContext.save.playerData.health;
+        fileChooseCtx->unk_24454[fileChooseCtx->unk_244A6] = gSaveContext.save.inventory.dungeonKeys[9];
+        fileChooseCtx->unk_24444[fileChooseCtx->unk_244A6] = gSaveContext.save.inventory.questItems;
+        fileChooseCtx->unk_24458[fileChooseCtx->unk_244A6] = gSaveContext.save.time;
+        fileChooseCtx->unk_24460[fileChooseCtx->unk_244A6] = gSaveContext.save.day;
+        fileChooseCtx->unk_24468[fileChooseCtx->unk_244A6] = gSaveContext.save.isOwlSave;
+        fileChooseCtx->rupees[fileChooseCtx->unk_244A6] = gSaveContext.save.playerData.rupees;
+        fileChooseCtx->unk_24474[fileChooseCtx->unk_244A6] =
+            (gSaveContext.save.inventory.upgrades & gUpgradeMasks[4]) >> gUpgradeShifts[4];
+
+        for (maskCount = i = 0; i < ARRAY_COUNT(gSaveContext.save.inventory.masks); i++) {
+            if (gSaveContext.save.inventory.masks[(s32)i] != ITEM_NONE) {
+                maskCount++;
+            }
+        }
+
+        fileChooseCtx->unk_24478[fileChooseCtx->unk_244A6] = maskCount;
+        fileChooseCtx->unk_2447C[fileChooseCtx->unk_244A6] =
+            (gSaveContext.save.inventory.questItems & 0xF0000000) >> 0x1C;
+    }
+
+    gSaveContext.save.time = D_801F6AF0;
+    gSaveContext.unk_3F3F = D_801F6AF2;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80146628.s")
+#endif
+
+void func_80146AA0(GameState* gameState, SramContext* sramCtx) {
+    s32 phi_v0;
+    u16 i;
+    FileChooseContext* fileChooseCtx = (FileChooseContext*)gameState;
+    s16 phi_a0;
+
+    if (gSaveContext.unk_3F3F) {
+        Sram_InitNewSave();
+        if (fileChooseCtx->unk_24480 == 0) {
+            gSaveContext.save.cutscene = 0xFFF0;
```

Comment:
Oh how odd, this is an s32 but other cutscene numbers are s16... this makes it a pain to decide whether to use 0xFFFx or -x for them...